tests/Functional/Action/Lti/GetUserAssignmentLtiLinkActionTest.php

```diff
@@ -394,5 +408,64 @@ static function (array $record) use ($assignment, $ltiRequestInResponse) {
             },
             Logger::INFO
         );
+
+        Carbon::setTestNow();
+    }
+
+    /**
+     * @SuppressWarnings(PHPMD.Superglobals)
+     */
+    public function testItReturnsLti1p3Link(): void
+    {
+        $this->generateKeyPairs();
+
+        Carbon::setTestNow(Carbon::create(2019, 1, 1, 0, 0, 0, new DateTimeZone('UTC')));
+
+        /** @var UserRepository $userRepository */
+        $userRepository = $this->getRepository(User::class);
+        $user = $userRepository->findByUsernameWithAssignments('user1');
+
+        $this->logInAs($user, $this->kernelBrowser);
+
+        $_ENV['LTI_VERSION'] = LtiRequest::LTI_VERSION_1P3;
+
+        $this->kernelBrowser->request('GET', '/api/v1/assignments/1/lti-link');
+
+        self::assertSame(Response::HTTP_OK, $this->kernelBrowser->getResponse()->getStatusCode());
+
+        $responseData = json_decode($this->kernelBrowser->getResponse()->getContent(), true, 512, JSON_THROW_ON_ERROR);
+
+        self::assertStringContainsString(
+            'iss=https://simple-roster.docker.localhost/platform',
+            urldecode($responseData['ltiLink'])
+        );
+        self::assertStringContainsString(
+            'login_hint=user1',
+            urldecode($responseData['ltiLink'])
+        );
+        self::assertStringContainsString(
+            'target_link_uri=http://localhost:8888/tool/launch',
+            urldecode($responseData['ltiLink'])
+        );
+        self::assertStringContainsString(
+            'lti_deployment_id=1',
+            urldecode($responseData['ltiLink'])
+        );
+        self::assertStringContainsString(
+            'client_id=demo',
+            urldecode($responseData['ltiLink'])
+        );
+
+        self::assertSame(LtiRequest::LTI_VERSION_1P3, $responseData['ltiVersion']);
+        self::assertSame([], $responseData['ltiParams']);
```

Comment:
This is not really nice. Let's decode the **full** response and assert on it like in the other test cases above please.

Reply:
There is no way to do it since we are generating different keys. I think it will also generate different values even with the same key pair,  that's why I'm checking parts of the link